test/libsolidity/smtCheckerTests/types/struct/struct_recursive_indirect_2.sol

```diff
@@ -54,6 +54,12 @@ contract C {
 // Warning 8364: (248-250): Assertion checker does not yet implement type struct C.S storage ref
 // Warning 8364: (248-255): Assertion checker does not yet implement type struct C.T storage ref
 // Warning 8364: (248-260): Assertion checker does not yet implement type struct C.S storage ref
+// Warning 6368: (183-190): CHC: Out of bounds access happens here.\nCounterexample:\n\n\nTransaction trace:\nC.constructor()\nC.f()
+// Warning 6368: (203-210): CHC: Out of bounds access happens here.\nCounterexample:\n\n\nTransaction trace:\nC.constructor()\nC.f()
+// Warning 6368: (230-237): CHC: Out of bounds access happens here.\nCounterexample:\n\n\nTransaction trace:\nC.constructor()\nC.f()
+// Warning 6368: (230-242): CHC: Out of bounds access happens here.\nCounterexample:\n\n\nTransaction trace:\nC.constructor()\nC.f()
+// Warning 6368: (248-255): CHC: Out of bounds access happens here.\nCounterexample:\n\n\nTransaction trace:\nC.constructor()\nC.f()
+// Warning 6368: (248-260): CHC: Out of bounds access happens here.\nCounterexample:\n\n\nTransaction trace:\nC.constructor()\nC.f()
 // Warning 6328: (223-263): CHC: Assertion violation happens here.\nCounterexample:\n\n\nTransaction trace:\nC.constructor()\nC.f()
```

Comment:
`struct storage` itself is fine, the problem is when there are assignments to storage pointers or references. Since we don't track aliasing, we remove all knowledge regarding pointers or references that might be pointing to the same as the modified area, that's why we see such false positives. Also recursive structs are not supported at all, so pretty much anything regarding recursive structs will give false positives